README: Update description of Musl support

razvand opened the pull request:
The contents of the `README.md` file are not up-to-date. Replace them with a short description of Musl, and point out that Musl is the default Unikraft libc.

razvand committed:
README: Update description of Musl support

The contents of the README.md file are not up-to-date. Replace them with
a short description of Musl, and point out that Musl is the default
Unikraft libc.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

kubanrob commented on README.md:
```diff
@@ -1,29 +1,14 @@
-## Build
+Musl is the default and recommended libc for Unikraft.
+[`newlib`](https://github.com/unikraft/lib-newlib) is another libc available for Unikraft.
+`newlib` also requires [`pthread-embedded`](https://github.com/unikraft/lib-pthread-embedded/), while Musl doesn't.
```
This looks not completely correct to me.

As far as I understand: `lib-newlib` does not support pthreads (`newlib` might do), so `lib-pthread-embedded` is required to use pthreads with `lib-newlib`. The other way around, `lib-pthread-embedded` is not compatible with `lib-musl`.

kubanrob replied:
At this point it might be good also mentioning that `nolibc` provides a minimal subset of libc required for the kernel.

kubanrob reviewed (changes requested):
Looks good, however I am not sure if the information about `lib-newlib` is correct.

razvand commented:
> Looks good, however I am not sure if the information about `lib-newlib` is correct.

Thanks, @kubanrob. I updated the README addressing your comments.

mariasfiraiala commented on README.md:
```diff
@@ -1,29 +1,16 @@
+[`newlib`](https://github.com/unikraft/lib-newlib) is another libc available for Unikraft.
+Note that networking support [`lwip`](https://github.com/unikraft/lib-lwip) is required.
+Musl provides thread support natively, while `newlib` requires [`pthread-embedded`](https://github.com/unikraft/lib-pthread-embedded/);
+Musl doesn't require (and must not be built with) `pthread-embedded.
```
```suggestion
Musl doesn't require (and must not be built with) `pthread-embedded`.
```

mariasfiraiala reviewed (changes requested):
Small comment on my part (you'll find the suggestion below). Also, I wonder whether it's worth mentioning the `clang` - `compiler-rt` - `__muldc3` issue. Even though not many people use `clang`, it's good for documentation purposes

razvand commented:
> Small comment on my part (you'll find the suggestion below). Also, I wonder whether it's worth mentioning the `clang` - `compiler-rt` - `__muldc3` issue. Even though not many people use `clang`, it's good for documentation purposes

@mariasfiraiala , done.

mariasfiraiala reviewed (approved):
Looks ok, thanks @razvand!

Reviewed-by: Maria Sfiraiala <maria.sfiraiala@gmail.com>

eduardvintila commented on README.md:
```diff
@@ -1,29 +1,23 @@
-## Build
+Musl is the default and recommended libc for Unikraft.
+[`newlib`](https://github.com/unikraft/lib-newlib) is another libc available for Unikraft.
+Note that networking support [`lwip`](https://github.com/unikraft/lib-lwip) is required.
```
```suggestion
Note that [`lwip`](https://github.com/unikraft/lib-lwip) is required for networking support.
```

One could understand that networking support is mandatory for all musl builds, which obviously is not the case.

eduardvintila reviewed (changes requested):
Hey, @razvand. Looks good, thanks for this. There is only one sentence which I would have it rephrased.

kubanrob commented:
Thank you @razvand!

Reviewed-by: Robert Kuban <robert.kuban@opensynergy.com>

eduardvintila reviewed (approved):
Thanks, @razvand!

Approved-by: Eduard Vintilă <eduard.vintila47@gmail.com>
